Fix background color of the `ShowMoreCard` component

xlisachan opened the pull request:
Fixes #9311 

The background color of the buttons in the ShowMoreCard and Card component have been updated. Please let me know your thoughts. Thanks!

ShowMoreCard component -
Before
<img width="999" alt="Screen Shot 2020-03-31 at 1 06 24 PM" src="https://user-images.githubusercontent.com/28129806/78055543-b0901800-7351-11ea-91fa-612d4565bafc.png">

After
<img width="994" alt="Screen Shot 2020-03-31 at 12 56 59 PM" src="https://user-images.githubusercontent.com/28129806/78055508-a4a45600-7351-11ea-8ab9-ed0b10911f46.png">

Rate your experience -
Before
<img width="350" alt="Screen Shot 2020-03-31 at 1 06 12 PM" src="https://user-images.githubusercontent.com/28129806/78055586-c1408e00-7351-11ea-98b1-8f9867cc50a7.png">

After
<img width="365" alt="Screen Shot 2020-03-31 at 12 56 27 PM" src="https://user-images.githubusercontent.com/28129806/78055608-c7366f00-7351-11ea-91ba-cfbf23d9d6a3.png">

willdurand commented on src/amo/pages/Addon/styles.scss:
```diff
@@ -188,3 +188,9 @@
+
+.AddonReviewCard-container {
+  footer {
+    background: rgba(10, 132, 255, 0);
```
Dont we have a variable with this color already?

xlisachan commented:
@willdurand, I moved the style updates into the correct stylesheets (AddOnReviewCard and RatingManager), as the updates were for `.AddonReviewCard-reply` and `.RatingManager`.

There is a variable for blue (`$blue-50`), but it is set to appear lighter in both sections (`background-color: transparentize($blue-50, 0.95)`). So instead I set the footers of each section to have a transparent background. Per the screenshot below, I also fixed the fading as requested.

<img width="359" alt="Screen Shot 2020-04-01 at 3 57 29 PM" src="https://user-images.githubusercontent.com/28129806/78183113-61bbae80-7435-11ea-82ce-696b26ad04c9.png">

<img width="994" alt="Screen Shot 2020-04-01 at 3 57 22 PM" src="https://user-images.githubusercontent.com/28129806/78183125-654f3580-7435-11ea-8fe7-5d36c3703159.png">

I initally felt a border color was needed as well, but thought it was a cleaner look with just the fade indicating that there was more text. Please see mockups below and let me know if you envisioned it differently.

<img width="347" alt="Screen Shot 2020-04-01 at 4 36 47 PM" src="https://user-images.githubusercontent.com/28129806/78184274-32a63c80-7437-11ea-8ccf-7b4e2cd6fa52.png">

<img width="996" alt="Screen Shot 2020-04-01 at 4 37 01 PM" src="https://user-images.githubusercontent.com/28129806/78184287-36d25a00-7437-11ea-91e5-94bf649316e6.png">

In terms of creating a story, this is my first time using Storybook and I don't think the aria-label carried through props. Can you please take a look and provide some guidance? Thank you!

willdurand commented on src/ui/components/ShowMoreCard/index.js:
```diff
@@ -129,21 +130,26 @@ export class ShowMoreCardBase extends React.Component<InternalProps> {
     const readMoreLink = (
-      <a
-        aria-label={i18n.gettext('Expand to read more')}
+      <Button
```
This is a bigger change. Can we do without it for now?

xlisachan replied:
I figured it out! I had set it to `props.ariaLabel` instead of just `ariaLabel`.

willdurand replied:
Can we not use `Button` in this patch?

xlisachan commented on src/amo/components/AddonReviewCard/styles.scss:
```diff
@@ -123,6 +123,14 @@
   }
+
+  .ShowMoreCard-contents::after {
+    background: linear-gradient(rgba(255, 255, 255, 0), #f4f9ff);
```
It's noted in the ShowMoreCard stylesheet that for the linear-gradient `rgba(255, 255, 255, 0)` is used instead of `transparent` to prevent a gray line in Safari.

The variable with the lightest shade of blue is $blue-50.  Can I create a variable that is set to `#f4f9ff`?

xlisachan commented:
Thanks for the feedback! 

Please see my comments above. I have made the following updates:
- Resolved the issue of aria-label not appearing (removed `props.`)
- Created a variable `gradient-color` set to `#f4f9ff`

Also, where did we land on the button's border-color - do you prefer them with or without a border?

Please review when you have a moment. Thanks again!

willdurand commented:
> Also, where did we land on the button's border-color - do you prefer them with or without a border?

I think we have to keep the border.

willdurand commented on src/amo/components/RatingManager/styles.scss:
```diff
@@ -27,6 +27,10 @@
   }
+
+  .ShowMoreCard-contents::after {
+    background: linear-gradient(rgba(255, 255, 255, 0), #f4f9ff);
```
I suppose we can reuse `$gradient-color` here, no?

willdurand commented on src/core/css/inc/vars.scss:
```diff
@@ -16,6 +16,7 @@ $button-background-color: #0095dd;
 $base-color: #fff;
 $link-color: $blue-60;
 $warning: #d63920;
+$gradient-color: #f4f9ff;
```
This should probably have a more specific name, like `showmorecard-gradient-color` or something.

xlisachan commented:
Please see the revisions below: 
- Reverted `Button` back to `a` tag in ShowMoreCard component; removed Read More from Button Storybook, Button component/stylesheet
- Revised variable from `gradient-color` to `showmorecard-gradient-color` and `RatingManager` stylesheet
- Added border to Read More links

willdurand commented on src/ui/components/ShowMoreCard/styles.scss:
```diff
@@ -36,6 +36,11 @@
   }
 }
 
+.ShowMoreCard-expand-link {
```
I am going to remove this actually. It does not look nice (I did not really pay enough attention in the screenshots, sorry!)
